src/bundles/plotly/sound_functions.ts

```diff
@@ -0,0 +1,99 @@
+import { context } from 'js-slang/moduleHelpers';
+import Plotly, { type Data, type Layout } from 'plotly.js-dist';
+import { get_duration, get_wave, is_sound } from '../sound';
+import { type Sound } from '../sound/types';
+import { CurvePlot, type DrawnPlot } from './plotly';
+
+const FS: number = 44100; // Output sample rate
+
+const drawnPlots: (DrawnPlot | CurvePlot)[] = [];
+context.moduleContexts.plotly.state = {
+  drawnPlots,
+};
+
+/**
+ * Visualizes the sound on a 2d line graph
+ * @param sound the sound which is to be visualized on plotly
+ */
+export function draw_sound_2d(sound: Sound) {
+  if (!is_sound(sound)) {
+    throw new Error(`play is expecting sound, but encountered ${sound}`);
+    // If a sound is already playing, terminate execution.
+  } else if (get_duration(sound) < 0) {
+    throw new Error('play: duration of sound is negative');
+  } else {
+    // Instantiate audio context if it has not been instantiated.
+
+    // Create mono buffer
+    const channel: number[] = [];
+    const time_stamps: number[] = [];
+    const len = Math.ceil(FS * get_duration(sound));
+
+    let temp: number;
+    let prev_value = 0;
+
+    const wave = get_wave(sound);
+    for (let i = 0; i < len; i += 1) {
+      temp = wave(i / FS);
+      time_stamps[i] = i / FS;
+      // clip amplitude
+      // channel[i] = temp > 1 ? 1 : temp < -1 ? -1 : temp;
+      if (temp > 1) {

```

Comment:
I think amplitude should not be clipped for draw_sound. That way if clipping happens, one can use this tool to see the original wave function.

Reply:
Clipping was done in this case as well, to prevent the graph from getting too zoomed out due to an unnecessarily large output for one of the timestamps since plotly will initially want to show all the data points. That being said your point is correct, since it will be a good way of seeing why was the audio not the way the user expected it to be.